c/blake3.h

```diff
@@ -69,6 +75,18 @@ BLAKE3_API void blake3_hasher_init_derive_key_raw(blake3_hasher *self, const voi
                                                   size_t context_len);
 BLAKE3_API void blake3_hasher_update(blake3_hasher *self, const void *input,
                                      size_t input_len);
+#if defined(BLAKE3_USE_LLFIO)
+BLAKE3_API void blake3_hasher_update_mmap(blake3_hasher *self,
+                                          const char *path) BLAKE3_NOEXCEPT;
+#endif // BLAKE3_USE_LLFIO
+#if defined(BLAKE3_USE_TBB)
+BLAKE3_API void blake3_hasher_update_tbb(blake3_hasher *self, const void *input,
+                                         size_t input_len);
+#if defined(BLAKE3_USE_LLFIO)
+BLAKE3_API void blake3_hasher_update_mmap_tbb(blake3_hasher *self,
+                                              const char *path) BLAKE3_NOEXCEPT;
```

Comment:
`blake3_hasher_update_mmap` and `blake3_hasher_update_mmap_tbb` should probably be able to report IO errors. Is it possible to get an `errno`-compatible `int` here?

Reply:
Also what's the right way to express a path in a C API that allows for UTF-16 paths on Windows?

Reply:
We could do that.

The way it's handled right now is that `blake3_hasher_update_mmap_base` will set `errno` if an error occurs: https://github.com/BLAKE3-team/BLAKE3/blob/fac71c9e6748e61e0c280e4e905f8f808b3c2ae8/c/blake3_llfio.cpp#L88

In the example I check for that by clearing `errno` before the call and checking it afterwards: https://github.com/BLAKE3-team/BLAKE3/blob/fac71c9e6748e61e0c280e4e905f8f808b3c2ae8/c/example-mmap.c#L25-L27

BurningEnlightenment commented earlier that we could return a bool to be more informative regarding whether an error occurred (so it's not necessary to do the clear manually first).

But an alternative would be to just return the code as an `int`. Is that what you would prefer?

Reply:
> Also what's the right way to express a path in a C API that allows for UTF-16 paths on Windows? Would we have to take some monstrosity like this?
> 
> ```
> typedef struct {
>     uint16_t* utf16_path;  // Windows
>     char* utf8_path;       // POSIX
> } FilePath;
> ```

Something like that I think, but if we want to be rigorous, we'd probably need to use `wchar_t` for Windows and `char8_t` for POSIX. But `char8_t` requires C23.

Reply:
> But an alternative would be to just return the code as an int. Is that what you would prefer?

Oh nice, yeah that sounds great.